### PR TITLE
filter Unit test

### PR DESCRIPTION
**Pull request by Philip-21**
**Notes for Reviewers**

This PR fixes  for issue #8087 
It contains unit test for the [filter.go](https://github.com/meshery/meshery/blob/master/mesheryctl/internal/cli/root/filter/filter.go) in mesheryctl

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Philip-21** [comment]
@leecalcote @alphaX86    @nebula-aac @hexxdump @abdullah1308 
i followed the mesheryctl's plan of mocking unit test from this [discussion](https://discuss.layer5.io/t/how-to-write-unit-integration-tests-for-mesheryctl/285), 
The tests coverage keeps failing . The error is shown [here](https://github.com/meshery/meshery/actions/runs/5506634388/jobs/10035585290?pr=8111) , i'm still figuring it out but i am just giving you all a heads up on it ,i would also appreciate some help .

**Comment by alphaX86** [comment]
@Philip-21 some tests won't appear on CodeCov if they are written very new. So, one way to check this is by unit testing the programs in VSCode which has inbuilt test features there.

**Comment by abdullah1308** [comment]
@Philip-21 Hey I have a PR making a few changes in `mesheryctl filter` here - https://github.com/meshery/meshery/pull/8063. 

Once that is merged, make sure these tests still work.

**Comment by Chadha93** [comment]
@Philip-21  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by Philip-21** [comment]
> @Philip-21 Hey I have a PR making a few changes in `mesheryctl filter` here - #8063.
> 
> Once that is merged, make sure these tests still work.

I've looked at the pr adding support for WASM filters is a great idea, it seems it  will  tackle the _**Error processing JSON response from server**_,  because i also ran the other testcases locally, they failed and gave the same error .

**Comment by Philip-21** [comment]
ci-checks
My Ci checks have passed on codecov with these latest commit, i added more  testcases would like reviews on this .

**Comment by Philip-21** [comment]
@leecalcote ,@abdullah1308 , @Chadha93 👀👀👀

**Comment by leecalcote** [comment]
@hexxdump do you mind offering review here? 🙏

File: mesheryctl/internal/cli/root/filter/filter_test.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		URL              string
+		Fixture          string
+		Token            string
+		ExpectedResponse string
+		ExpectHelp       bool
+		ExpectErr        bool
+	}{
+		{
+			Name:             "filter viewcmd with name",
+			Args:             []string{"view", "view-filter-name"},
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/filter/filter_test.go:43]: can we add tests to use `filter viewcmd` with valid name and some non-existing filtername?

File: mesheryctl/internal/cli/root/filter/filter_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+			Token:            filepath.Join(fixturesDir, "token.golden"),
+			ExpectedResponse: "filter.id.view.output.golden",
+			Fixture:          "filter.name.view.api.response.golden",
+			URL:              testContext.BaseURL + "/api/filter",
+			ExpectHelp:       false,
+			ExpectErr:        false,
+		},
+		{
+			Name:             "filter viewcmd with ID",
+			Args:             []string{"view", "c0c6035a-b1b9-412d-aab2-4ed1f1d51f84"},
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/filter/filter_test.go:53]: can we have a test for invalid ID as well?

File: mesheryctl/internal/cli/root/filter/filter_test.go (hunk 1 above)
Comment (Philip-21) [reply mesheryctl/internal/cli/root/filter/filter_test.go:43]: Done

File: mesheryctl/internal/cli/root/filter/filter_test.go (hunk 2 above)
Comment (Philip-21) [reply mesheryctl/internal/cli/root/filter/filter_test.go:53]: Done

**Review by hexxdump — APPROVED** [review]
LGTM

**Comment by Philip-21** [comment]
@leecalcote everything is good with this Pr.

**Pull request merged**